index.js

```diff
@@ -50,7 +50,9 @@ module.exports = {
   },
 
   cacheKeyForTree(treeType) {
-    return calculateCacheKeyForTree(treeType, this, this.package.root);
+    const paths = this.package.paths || this.package.treePaths;
```

Comment:
Can you explain these changes a bit so I understand what's going on.

Reply:
Sure. 

When you boot the dummy app, the `package.root` for the engine instance is the same as the dummy app instance: the engine `root` folder.

With this change, the tree cache key changes by the tree folder for the instance. So, for example, for the public tree, the engine cache key will be `root/public` and the dummy app cache key will be `root/tests/dummy/public`.

When used inside a real app, in the same example as before, the engine cache key for public tree will be `root/node_modules/engine/public` and for the app `root/public`.

Reply:
Thanks!  Curious, is `paths` ever falsey?

Reply:
Yep, When instance is an engine there is a `paths` property and when it is the app it has a `treePaths` property 🤷‍♂ 

Also, some addons has no property at all, so I use the same `root` property.